src/main/java/graphql/schema/diffing/DiffImpl.java

```diff
@@ -390,6 +394,7 @@ private double calcLowerBoundMappingCost(Vertex v,
                 anchoredVerticesCost++;
             }
 
+            runningCheck.check();
```

Comment:
I added more checks to places that the algorithm would get stuck on when the changes were large.